packages/smart-wallet/src/walletFactory.js

```diff
@@ -237,6 +248,19 @@ export const prepare = async (zcf, privateArgs, baggage) => {
    */
   const makeSmartWallet = prepareSmartWallet(baggage, shared);
 
+  // One time repair for incarnation 2. We're adding WatchedPromises to allow
+  // wallets to durably monitor offer outcomes, but wallets that already exist
+  // need to be backfilled. This code needs to run once at the beginning of
+  // incarnation 2, and then shouldn't be needed again.
+  if (!baggage.has(UPGRADE_TO_INCARNATION_TWO)) {
+    trace('Wallet Factory upgrading to incarnation 2');
+
+    for (const wallet of walletsByAddress.values()) {
```

Comment:
perhaps worth commenting that this can take a long while, and we expect it to run during an upgrade handler, between blocks